spec/ruby/core/dir/fchdir_spec.rb

```diff
@@ -0,0 +1,78 @@
+require_relative '../../spec_helper'
+require_relative 'fixtures/common'
+
+ruby_version_is '3.3' do
+  has_fchdir = begin
+    dir = Dir.new('.')
+    Dir.fchdir(dir.fileno)
+    true
+  rescue NotImplementedError
+    false
+  rescue Exception
+    true
```

Comment:
Should it be `false` here? If it raises an exception it's not supported, right?

Reply:
I changed it to false in ruby/spec: https://github.com/ruby/spec/commit/c075975276bce25ea1edc60fa4eefd6156a3107c

Reply:
If it isn't supported, it should be raising NotImplementedError.  If it raises another exception, that indicates it is supported, but the support is broken, which means the specs should rightfully fail for it. 

I don't like the change in https://github.com/ruby/spec/commit/c075975276bce25ea1edc60fa4eefd6156a3107c, because it allows a ruby implementation that advertises Ruby 3.3 version to not implement the method.  If a ruby implementation says it implements Ruby 3.3, it must implement the method, even if the method always raises NotImplementedError.

Reply:
(the trick is `respond_to?` returns false for a method defined as `rb_notimplement`)

Reply:
That works for me, thanks!